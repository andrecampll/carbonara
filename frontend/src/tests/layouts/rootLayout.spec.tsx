import { cleanup, screen } from '@testing-library/react'
import { ReactNode } from 'react'
import { afterEach, describe, expect, it, vitest } from 'vitest'

import RootLayout from '@/app/layout'
import { render } from '@/utils/tests/render'

vitest.mock('@apollo/experimental-nextjs-app-support/ssr', () => ({
  __esModule: true,
  ApolloNextAppProvider: ({ children }: { children: ReactNode }) => {
    return children
  },
}))

vitest.mock('next/font/google', () => ({
  __esModule: true,
  Inter: () => {
    return {
      subsets: ['latin'],
      variable: '--font-inter',
    }
  },
}))

afterEach(cleanup)

describe('RootLayout', () => {
  it('should be able to render RootLayout correctly', () => {
    const { container } = render(<RootLayout>Testing RootLayout</RootLayout>)

    expect(screen.getByText('Testing RootLayout'))
    expect(container.firstChild).toMatchSnapshot()
  })
})
