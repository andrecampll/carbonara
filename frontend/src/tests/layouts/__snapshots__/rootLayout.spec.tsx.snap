// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RootLayout > should be able to render RootLayout correctly 1`] = `
<html
  class="--font-inter"
  lang="en"
>
  <body
    class="undefined bg-background antialiased"
  >
    <header
      class="bg-gradient-to-r from-primary-500 to-secondary-500 p-4 md:px-12"
    >
      <div
        class="m-auto flex max-w-5xl items-center justify-between"
      >
        <a
          href="/"
        >
          <div
            class="flex items-center gap-2"
          >
            <span
              class="text-xl font-bold text-white"
            >
              Carbonara
            </span>
          </div>
        </a>
        <div
          class="flex items-center gap-4"
        >
          <p
            class="text-white"
          >
            John Doe
          </p>
          <div
            class="rounded-full bg-white p-1"
          >
            <span
              class="font-bold text-black"
            >
              JD
            </span>
          </div>
        </div>
      </div>
    </header>
    <main
      class="mb-8 min-h-[84vh] px-4 md:px-12"
    >
      <div
        class="m-auto max-w-5xl"
      >
        Testing RootLayout
      </div>
    </main>
  </body>
</html>
`;
